config/src/lib.rs

```diff
@@ -377,11 +378,21 @@ impl Committee {
         (total_votes + 2) / 3
     }
 
-    /// Returns a leader node in a round-robin fashion.
+    /// Returns a leader node as a weighted choice seeded by the provided integer
     pub fn leader(&self, seed: usize) -> PublicKey {
-        let mut keys: Vec<_> = self.authorities.keys().cloned().collect();
-        keys.sort();
-        keys[seed % self.size()].clone()
+        let mut seed_bytes = [0u8; 32];
+        seed_bytes[32 - 8..].copy_from_slice(&seed.to_le_bytes());
```

Comment:
why `32-8`?

Reply:
The seed is a u64 (8 bytes)
Ah, yes, you're right.We should include the epoch in there.

Reply:
got it